### ISSUE TITLE
List headers are disabled

### ISSUE THREAD

**Issue by jmbarbier** [nodemailer/nodemailer#751]
It seems that List headers are not set using `list` parameter. Some code digging shows that

```js
    setListHeaders() {
        if (!!this.message || !this.data.list || typeof this.data.list !== 'object') {
            return;
        }
```
from [lib/mailer/mail-message.js](https://github.com/nodemailer/nodemailer/blob/1ceab54918107013ee8e81067f81c66917444e24/lib/mailer/mail-message.js#L155) always returns so the addition of list headers is skipped.

If we re-enable list headers setting with `if (!this.message ...)` instead of `if (!!this.message ...)` we face another problem of `this.getListHeaders(..)` not existing [lib/mailer/mail-message.js#160](https://github.com/nodemailer/nodemailer/blob/1ceab54918107013ee8e81067f81c66917444e24/lib/mailer/mail-message.js#L160). It is now `this._getListHeaders(..)`. 

Making those 2 small corrections makes list-headers work again.

Maybe there is a reason for that, but in that case documentation should be updated (by the way, there are no tests for lists-xxx headers)

1. Nodemailer version you are having problems with (eg. v3.1.7)
2. Node.js version you are using (v6.10.0)
3. Your operating system (Ubuntu 16.04 etc.)
4. Nodemailer configuration settings = from documentation, list settings sample
5. If possible, include a minimal test case that can be used to verify your issue

**Comment by andris9** [comment]
Thanks, you're right, I'll take a look at this

**Comment by andris9** [comment]
Fixed it and released as v3.1.8
